### ISSUE TITLE
API: Allow IGC file downloads

### ISSUE THREAD

**Issue by Turbo87** [Turbo87/ogn-web-gateway#11]
We could query the database for all records of a certain ID and output it as an IGC file.

**Comment by estk** [comment]
I'm looking into this, as you suggested re our email conversation :)

**Comment by estk** [comment]
Ok, after a bit of consideration, I feel like a good place to start will be to extend https://github.com/Turbo87/rust-igc to also serialize IGC, then depend on that to create the file. How does that approach sound?

**Comment by Turbo87** [comment]
well, now that you've found the other IGC parser... :D

due to the low-level nature of `igc-rs` it might be quite a bit easier to use that to write IGC files. I've opened https://github.com/Joey9801/igc-rs/issues/6 to discuss the possibility of adding writing capabilities to that crate.

**Comment by estk** [comment]
Yeah sounds good, additionally I was wondering why in rust-igc you didn't use serde to do the seralization/deserialization?

**Comment by Turbo87** [comment]
from my understanding serde is meant more for (de)serialization of generic data (strings, numbers, lists, maps, etc.), but in this case the file has a fixed format, that does not seem to fit very well with using something like serde.

I also tried several of the more fancy parsing libraries but in the end using `regex` seemed easiest. I noticed though that `igc-rs` is just using plain string processing, which is probably also good enough. 

The other main difference currently is that `rust-igc` is based on byte arrays, while `igc-rs` operates on strings. Technically IGC files have to be ASCII, but nobody really cares apparently and there are plenty of files around that have non-ASCII characters. Most seem to be UTF8, but there are definitely also some that use Latin1 instead.
